src/app/teacher/teacher-run-list/teacher-run-list.component.ts

```diff
@@ -212,8 +200,51 @@ export class TeacherRunListComponent implements OnInit {
   private highlightNewRun(runId: number): void {
     for (const run of this.runs) {
       if (run.id === runId) {
-        run.isHighlighted = true;
+        run.highlighted = true;
       }
     }
   }
+
+  private unselectAllRuns(): void {
+    for (const run of this.runs) {
+      run.selected = false;
+    }
+  }
+
+  protected selectRunsOptionChosen(value: string): void {
+    this.filteredRuns.forEach((run: TeacherRun) => {
+      switch (value) {
+        case 'all':
+          run.selected = true;
+          break;
+        case 'none':
+          run.selected = false;
+          break;
+        case 'completed':
+          run.selected = run.isCompleted(this.configService.getCurrentServerTime());
+          break;
+        case 'running':
+          run.selected = run.isActive(this.configService.getCurrentServerTime());
+          break;
+        case 'scheduled':
+          run.selected = run.isScheduled(this.configService.getCurrentServerTime());
+          break;
+      }
+    });
+    this.runSelectedStatusChanged();
+  }
```

Comment:
Put the logic in TeacherRun?
```suggestion
  protected selectRunsOptionChosen(option: SelectRunsOption): void {
    const now = this.configService.getCurrentServerTime();
    this.filteredRuns.forEach((run: TeacherRun) => run.updateSelected(option, now));
    this.runSelectedStatusChanged();
  }
```